### ISSUE TITLE
VAE Over-splitting

### ISSUE THREAD

**Issue by btmartin721** [sokrypton/sp_deli#2]
Hello,

I used VAE for species delimitation as well as Random Forest and t-SNE. RF and t-SNE give me a more conservative optimal K of about 5-6, which seems reasonable to me. However, VAE gives me K=8. which is also found by BFD*, and is probably over-splitting in the species I'm working with. For VAE, I used the same dataset that I used for t-SNE and RF.

Do you have an idea as to why VAE might be over-splitting? Is there something I can change in the code to make the model more conservative? Or if not, can you think of any reason why VAE might have a tendency to over-split?

Thanks for your time.

**Comment by btmartin721** [comment]
OK so I think I've figured this out. Long story short, I had to do a lot of research to understand how to tune variational autoencoders.

I was underfitting the model by not running enough epochs. I subset 20% of the data to get 80%/20% training/validation datasets, then plotted the accuracy and loss for each to assess fit. The traces hadn't converged at 1000 epochs, which is what the default parameter was. So, I increased it to 10000 and it then was overfitting, but the elbow was at around ~2000 epochs. After re-running with 2K epochs, the clusters made way more sense.
